### PR TITLE
feat(firestore): Support overriding of service endpoint

### LINKED ISSUE #3757: [firestore] Support overriding endpoint host
Support overriding the endpoint host in the veneer layer, by setting either constructor argument or config field.

See #3752 for an example from bigtable.

### PR DESCRIPTION
**Pull request by quartzmo**
[closes #3757]

### CODE REVIEW

File: google-cloud-firestore/lib/google/cloud/firestore.rb [hunk 1]
```diff
@@ -72,12 +74,13 @@ module Firestore
...
       def self.new project_id: nil, credentials: nil, scope: nil, timeout: nil,
-                   client_config: nil, emulator_host: nil, project: nil,
-                   keyfile: nil
+                   client_config: nil, endpoint: nil, emulator_host: nil,
+                   project: nil, keyfile: nil
         project_id    ||= (project || default_project_id)
         scope         ||= configure.scope
         timeout       ||= configure.timeout
         client_config ||= configure.client_config
+        endpoint ||= configure.endpoint
```
Comment (dazuma) [inline google-cloud-firestore/lib/google/cloud/firestore.rb]: Can you either make the whitespace match the formatting around it, or remove the fancy formatting in the other lines?

File: google-cloud-firestore/test/google/cloud/firestore_test.rb [hunk 2]
```diff
@@ -18,7 +18,7 @@
   describe "#firestore" do
     it "calls out to Google::Cloud.firestore" do
       gcloud = Google::Cloud.new
-      stubbed_firestore = ->(project, keyfile, scope: nil, timeout: nil, client_config: nil) {
+      stubbed_firestore = ->(project, keyfile, scope: nil, timeout: nil, host: nil, client_config: nil) {
```
Comment (dazuma) [inline google-cloud-firestore/test/google/cloud/firestore_test.rb]: Check that passed host is nil

File: google-cloud-firestore/lib/google/cloud/firestore.rb (hunk 1 above)
Comment (quartzmo) [reply google-cloud-firestore/lib/google/cloud/firestore.rb]: Done!

File: google-cloud-firestore/test/google/cloud/firestore_test.rb (hunk 2 above)
Comment (quartzmo) [reply google-cloud-firestore/test/google/cloud/firestore_test.rb]: Done!

**Pull request merged**